pkg/provider/provider.go

```diff
@@ -46,3 +50,21 @@ type ResourceProvider interface {
 	IntrospectSummary() interface{}
 	ReconcileNode(nodeName string) bool
 }
+
+// GetWinWarmPoolConfig retrieves Windows warmpool configuration from ConfigMap, falls back to using default values on failure
+func GetWinWarmPoolConfig(log logr.Logger, w api.Wrapper, isPDEnabled bool) *config.WarmPoolConfig {
```

Comment:
Why should we have this shared function in the provider interface file?

Reply:
I have since migrated the function to a `util.go` file in the pool package.